### DISCUSSION TITLE
Provide an easier way to handle missing secrets

### DISCUSSION THREAD

**Discussion by jsoref** [community/community#12263] [Actions] [answered]
I create *lots* of forks and generally each fork I create will eventually gain a workflow of my creation as well as at least one push past that point.

Unfortunately, most repository maintainers don't understand the ramifications of using secrets, and the cost of running workflows that will inevitably fail when the secrets they require aren't available.

Here's the most recent failure:
<img width="989" alt="image" src="https://user-images.githubusercontent.com/2119212/156261750-2c54e741-4aec-4834-a2fb-4a4f1f9ba87a.png">
```
Run amondnet/vercel-action@v20
/home/runner/work/_actions/amondnet/vercel-action/v20/dist/index.js:2205
        throw new Error(`Input required and not supplied: ${name}`);
```
https://github.com/jsoref/view_components/runs/5383170775?check_suite_focus=true#step:10:7

What matters to me (here) is not the failure, it's the fact that the failure burned 6 minutes and one slot.

I have 20 concurrent slots for my busy org (yes, the above isn't from the org, but I believe my user account also has a similar limit).

Those slots are precious. Some repositories spend slots like there's no tomorrow. Whether they use `matrix` or just lots of parallel files (or both).

It'd be really nice if I didn't have to go to each repository and also make a bunch of PRs to make their workflows not spend lots of GitHub's Runner's cycles / my available slots.

Here's an example of the boilerplate that I end up writing:
https://patch-diff.githubusercontent.com/raw/primer/.github/pull/2.patch

It's a lot of code. And I write it a lot. I also end up making errors some non-trivial number of times. And of course, there's the review process -- someone has to review my code, make sure it's doing what it's supposed to (and not stealing the secret).

I understand that changes to the workflow definition can be seen as expensive, but I think it'd be pretty valuable for the workflow to be able to declaratively say that a job shouldn't run if a secret is missing.

(Examples of projects being justifiably paranoid are available upon request.)

**Comment by ethomson** [comment, marked as answer]
Interesting - thanks as usual for the feedback @jsoref. We'll put this on the backlog.

**Reply by jsoref** [reply]
Two examples of people trying to address this problem:
* https://github.com/github/docs/issues/16306#issuecomment-1076254205 implemented as: https://github.com/github/docs/commit/f54a2c3b27650cd384ca5615f2eef4d366a2086e
* https://github.com/robusta-dev/robusta/pull/227#discussion_r812194868 implemented as: https://github.com/robusta-dev/robusta/commit/2afd22fbb0ef05176fc80ea20c2c03886f3f55c7
